File: KIJUNG/PRO_WEEK_02/타겟넘버.cpp
#include <vector>

using namespace std;
int answer;

void dfs(int idx, int sum, vector<int> &numbers, int target){
    if(idx>=numbers.size()) {
        if(sum == target) answer++;
        return;
    }
    dfs(idx+1, sum+numbers[idx], numbers, target);
    dfs(idx+1, sum-numbers[idx], numbers, target);
}

int solution(vector<int> numbers, int target) {
    dfs(1,numbers[0],numbers,target);
    dfs(1,-numbers[0],numbers,target);
    
    return answer;
}

File: DONGUK/프로그래머스_WEEK 2/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    answer = 0
     
    q = deque()
    q.append((0, 0))
    
    while q:
        top = q.popleft()
        sum_, index = top[0], top[1]
        
        if index <= len(numbers) - 1:
            q.append((sum_ + numbers[index], index + 1))
            q.append((sum_ - numbers[index], index + 1))
        
        if index == len(numbers):
            if sum_ == target:
                answer += 1
        
    return answer